### DISCUSSION TITLE
How to turn off newly added feature?

### DISCUSSION THREAD

**Discussion by d3r3kr0b3rts** [community/community#113744] [Copilot Conversations]
### Select Topic Area

Question

### Body

There is a new feature that I think was introduced through Copilot which I would like to turn off. Wherever the cursor is, this feature is reading the word around the cursor, automatically finding all instances of the word, and then duplicating any changes I make at the cursor on the other instances of that word in the file. So if there are 3 variables named "$test" in my code and I make any changes to the characters in that variable name, something is automatically changes all other instances of the word "test" in the file (with or without a proceeding $). If there is a variable named "$test" and also an array with a key named "$array['test']" and I delete the word "test" it will delete the word "test" which comes after the $ in the variable name, and it will delete the word "test" in the array key $array['test']. Any help finding the setting to turn this feature off is appreciated.

An update: this only happens when I'm selecting a php variable or function name.

**Comment by AKASH-PRASAD7** [comment]
It seems like you’re referring to a feature that’s not part of Copilot, but rather a feature of your code editor. This feature is often called “multi-cursor editing” or “simultaneous editing”. It allows you to make changes to multiple instances of a selected word at the same time.

To disable the multi-cursor functionality in Visual Studio Code, you can follow these steps:

1. Open Visual Studio Code.
2. Click on the gear icon in the lower left corner to open the settings menu.
3. Select "Keyboard Shortcuts" from the menu.
4. In the search box, type "Add Cursor" to filter the commands.
5. For each command that adds a cursor and has a keyboard shortcut associated, right-click on the command and select "Remove Keybinding".
6. Also, search for "Multi-Cursor". It could return the command "Toggle Multi-Cursor Modifier" that is associated with a combination of modifier keys (one or more of Ctrl, Alt, Shift). 


I found a StackOverflow issue for the same-

https://stackoverflow.com/questions/53651080/disable-multi-cursor-functionality.

**Reply by d3r3kr0b3rts** [reply]
It's like that feature, but that feature requires some input other than just placing the cursor somewhere. For example, first I need would need to highlight the word to search for, then press a keyboard shortcut to highlight all other instances of the word.

Here I am not pressing any shortcuts to highlight the other instances of the word. Whenever I move the cursor anywhere in the file, something is automatically finding any other instances of whatever word the cursor lands on.

I've turned off "selection highlighting" and "semantic highlighting" in VS Code settings. Another thing is the highlighted words are a different color from the default selection highlighting. In my theme, the default VS Code highlight uses a transparent teal color to highlight other words. When I double click a word for example and highlight it, all other instances will highlight in teal.

But if I just place the cursor somewhere, without highlighting an entire word, all of the other instances of the word around the cursor will be highlighted in red instead of teal.

**Reply by d3r3kr0b3rts** [reply]
In this screenshot all I did was single click before the "m" in the first instance of the variable "$family" and pressed deleted and the "m" in both other instances of the variable name also deleted. I didn't do anything to select the other instances first I just single clicked on the page. There are no other cursors either. Usually Selection Highlighting creates a new cursor for every where that edits will occur and these edits occur without a cursor.

![Desktop Screenshot 2024 03 23 - 01 44 54 82](https://github.com/community/community/assets/13438829/241f2bf0-b2f4-4132-886b-f06d93d688a2)

**Reply by d3r3kr0b3rts** [reply]
For comparison this is what VS Code's feature "selection highlighting" looks like. Notice the highlighting is a different color and also there are two cursors.

![Desktop Screenshot 2024 03 23 - 01 53 21 77](https://github.com/community/community/assets/13438829/5974aefb-6956-434a-b08f-20168bdc77d4)

**Comment by d3r3kr0b3rts** [comment]
It was called "PHP linked variable editing" thanks for the help!
